src/main/java/org/galatea/starter/entrypoint/PriceRestController.java

```diff
@@ -31,15 +30,23 @@
 
   /**
    * Send the received text to the PriceService to be processed and send the result out
+   *
+   * @GetMapping to link http GET request to this method
+   * @RequestParam to take a parameter from the url
+   * @return
    */
-  // @GetMapping to link http GET request to this method
-  // @RequestParam to take a parameter from the url
+
+
   @GetMapping(value = "${webservice.halpath}", produces = {MediaType.APPLICATION_JSON_VALUE})
   public String priceEndpoint(
-      @RequestParam(value = "stock") String stock,
-      @RequestParam(value = "daysToLookBack", defaultValue= "1", required = false) int daysToLookBack,
-      @RequestParam (value = "requestId", required = false) String requestId) {
+
+  @RequestParam(value = "stock") String stock,
+      @RequestParam(value = "days", defaultValue= "1", required = false) String daysToLookBack,
+      @RequestParam (value = "requestId", required = false) String requestId)
+
+      throws IOException{
     processRequestId (requestId);
-    return priceService.processStock(stock, daysToLookBack);
-  }
+    priceService.getPricesByStock(stock, daysToLookBack);
+    return priceService.BuildMeta(stock, daysToLookBack);
```

Comment:
return a FullResponse object, not the string

Reply:
Had it return a string because I haven't been able to find a way to pretty-print the response on an Object